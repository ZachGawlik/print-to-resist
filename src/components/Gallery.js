import PropTypes from 'prop-types';
import React from 'react';
import ListingTile from './ListingTile';
import '../styles/Gallery.css';

class Gallery extends React.Component {
  componentDidMount() {
    this.props.getListings();
  }

  render() {
    const { listings } = this.props;
    return (
      <div className="Gallery" id="gallery">
        <h2>Spread the word!</h2>
        <p>Select a page to post around your city</p>
        <div>
          {listings.map(listing => (
            <div className="Gallery__tile" key={listing.listingId}>
              <ListingTile listing={listing} />
            </div>
          ))}
        </div>
      </div>
    );
  }
}

Gallery.propTypes = {
  getListings: PropTypes.func.isRequired,
  listings: PropTypes.array.isRequired
};

export default Gallery;
