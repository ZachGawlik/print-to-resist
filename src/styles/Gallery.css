.Gallery__tile {
  display: inline-block;
  width: 75%;
  padding: 10px;
}

@media (min-width: 555px) {
  .Gallery__tile {
    width: 50%;
  }
}

@media (min-width: 750px) {
  .Gallery__tile {
    width: 33.3333%;
  }
}
