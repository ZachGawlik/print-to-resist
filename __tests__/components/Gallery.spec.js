import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Gallery from '../../src/components/Gallery';

const listing97 = {
  listingId: 97,
  deadline: '2017-06-01T04:00:00.000Z',
  description: 'Upcoming dates for NYC congress town halls',
  instruction: '',
  isColor: false,
  paperSize: 'Letter',
  dateSubmitted: '2017-04-13T13:49:47.000Z',
  isApproved: 1,
  title: 'NYC Congress Town Halls',
  thumbnailId: 'c0c51fba8b26122d2c368abbfd5da8b6.png',
  imageIds: ['c0753ac7045056c7f73779a80db8ae55.png'],
  tags: ['nyc', 'town hall'],
  links: []
};

describe('Gallery', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Gallery
        galleryFilters={{
          activeOnly: false,
          colorOption: '',
          tag: null
        }}
        getListings={jest.fn()}
        listings={[listing97]}
        setColorFilter={jest.fn()}
        setTagFilter={jest.fn()}
        toggleActiveOnly={jest.fn()}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
