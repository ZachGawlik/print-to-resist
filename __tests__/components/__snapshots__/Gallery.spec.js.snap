// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders as expected 1`] = `
<div
  className="Gallery"
  id="gallery"
>
  <h2>
    Spread the word!
  </h2>
  <p>
    Select a page to post around your city
  </p>
  <FilterBar
    galleryFilters={
      Object {
        "activeOnly": false,
        "colorOption": "",
        "tag": null,
      }
    }
    setColorFilter={[Function]}
    setTagFilter={[Function]}
    toggleActiveOnly={[Function]}
  />
  <div>
    <div
      className="Gallery__tile"
    >
      <ListingTile
        deadline="2017-06-01T04:00:00.000Z"
        listingId={97}
        setTagFilter={[Function]}
        tags={
          Array [
            "nyc",
            "town hall",
          ]
        }
        thumbnailId="c0c51fba8b26122d2c368abbfd5da8b6.png"
        title="NYC Congress Town Halls"
      />
    </div>
  </div>
</div>
`;
